src/Extensions/ElementalAreasExtension.php

```diff
@@ -202,7 +202,11 @@ public function updateCMSFields(FieldList $fields)
 
             $editor = ElementalAreaField::create($eaRelationship, $area, $this->getElementalTypes());
 
-            if ($this->owner instanceof SiteTree && $fields->findOrMakeTab('Root.Main')->fieldByName('Metadata')) {
+            $insertBefore = Config::inst()->get(get_class($this->owner), 'insert_before_field_name');
+            
+            if ($insertBefore && $fields->dataFieldByName($insertBefore)) {
+                $fields->addFieldToTab('Root.Main', $editor, $insertBefore);
```

Comment:
What if your intended field wasn't in the `Root.Main` tab?
Using `$fields->insertBefore($insertBefore, $editor);` would be better I guess.

Reply:
Good point! I've updated it.